app/services/address_validation_service.rb

```diff
@@ -30,25 +30,38 @@ def po_box?(validation)
     #   does not normalize street1 so check case insensitive with optional periods and
     #   space, e.g. look for 'po box', 'P. O. Box'
     # https://www.easypost.com/errors-guide
+    false unless validation.present?
+
     validation.street1.match(/P\.?\s*O\.?\s*BOX/i) || (
       !validation.verifications.delivery.success &&
       validation.verifications.delivery.errors[0].code == 'E.BOX_NUMBER.INVALID'
     )
   end
 
+  def duplicate_unit?(validation)
+    # Check for duplicate apt/unit no on street lines 1 and 2
+    validation.present? && validation.street2.present? &&
+      validation.street1.end_with?(validation.street2)
```

Comment:
Is it guaranteed that validation.street1 will exist? What do you think about adding an optional chain operator after validation.street1 (e.g. validation.street1?.end_with...)? I'm also no ruby maestro so maybe that's not even possible.

Reply:
Good question. There is front end validation for street1 since it is a required field so unlikely that it would be submitted empty, but I can add just in case.